### DISCUSSION TITLE
Como retornar audio para o usuário?

### DISCUSSION THREAD

**Discussion by deppbrazil** [frontendbr/forum#706] [Discussão]
Vcs já ouviram falar de um `script ` que retorna audio para os usuários que necessitam de recursos de acessibilidade??

Sei que a melhoria no código e alguns atributos do html5 ajudam os leitores de telas a retornarem um audio mais fiel para o usuário, mas mesmo assim tem muitos leitores de telas que pegam divs e tags desnecessárias ficando um pouco confuso para quem necessita do recurso...

Existe alguma forma do tipo, eu passar ao script apenas a área que eu quero que ele retorne para o leitor? 

Ex: Em meu projeto tenho um `div ` `conteudo` no qual engloba todo o conteúdo para meu usuario final, tem alguma forma de eu pegar tudo que for `txt ` dentro dessa `div ` e retornar para o usuario em forma de audio??

**Comment by AmandaFonseca** [comment]
Olha eu já ouvi falar que se você usar algumas tags semânticas tipo  <main> <section> <nav> alguns leitores de telas ignoram MAAAASSSS não tenho como afirmar isso com certeza, nem testar pois estou cheia de trabalho ....

**Comment by felipefialho** [comment]
@reinaldoferraz @talitapagani

**Comment by bernardodiasc** [comment]
Existem alguns, mas geralmente são softwares proprietários e caros. Alguns sites de notícia ou por exemplo em artigos no site da editora O'Reilly há esse recurso de leitura do texto. Eu tinha feito uma pesquisa sobre esse tema alguns anos atrás, mas não consegui encontrar os links e informações, leitor de tela é algo bem interessante e útil de se ter. :)

**Comment by zemariobandeira** [comment]
Olá @deppbrazil , de fato existem muitas incógnitas relacionadas a boas práticas de desenvolvimento para acessibilidade na web, mas você pode seguir alguns passos simples e obter uma melhoria significativa no seu projeto.

1. Utilize alguma ferramenta de validação de acessibilidade na sua página: existem boas ferramentas na web (eu costumo recomendar a WAVE Web Accessibility Tool) mas você também pode encontrar extensões de navegador e outros tipos de ferramentas com o mesmo propósito;

2. Se você quer dizer ao software o significado de determinadas tags da sua página, o ideal é seguir as recomendações de semântica (a tag main poderia ser usada para o que você chama de conteúdo principal, por exemplo);

3. Outra maneira interessante seria controlar o fluxo de leitura do software leitor de tela fazendo o uso do atributo tabindex em suas tags;

4. Como conteúdo adicional, vale a pena dar uma olhada no recurso ARIA - Accessible Rich Internet Applications;

Valeu!

**Comment by deppbrazil** [comment]
@bernardodiasc @MarioDKR Show muito obrigado pelas dicas, já até utilzei algumas delas, o WAVE Web Acessibility é muito bom pois ele meio que avalia quanto seu código é sensível para leitores de telas, e o ARIA tb é mto bom...

Mas o que procuro mesmo seria um recurso de Screen Reader dentro da sua propria página html, ou seja sem a necessidade de utilizar o leitores de telas, sendo assim essa aplicação faria a leitura de determinadas divs de conteudos no caso poderia trabalhar com IDs e passar essas IDs ao script e ele retornar em forma de audio para usuarios que necessitam do recurso... sei que seria uma aplicação mais complexa...

Me disseram que o G1 tem um recurso parecido no qual retorna como audio algumas matérias no site quando ativado o recurso, mas eu não achei.... 

A [Sambatech ](http://sambatech.com/samba-play/) tem algo semelhante porém com libras, ou seja tem o player rodando o vídeo e acima do video um botão que chama a aplicação, quando requisitado um avatar reconhece os movimentos feitos no video e retorna em forma de libras com esse [avatar ](http://www.sambatech.com.br/prototipos/demo/ramon/weblibras/weblibras.html) para o usuario porém é um recurso que ainda trava um pouco... 

### Se alguém souber de algum recurso parecido porém para retornar audio para deficientes visuais compartilha ai.... 
## Valew! =) ##

**Comment by zemariobandeira** [comment]
@deppbrazil  eu entendi o que você quer...

Uma dica importante seria tentar validar com o usuário a necessidade de uma ferramenta assim.
Falo isso pois experienciei algo semelhante no ano passado.
Estive tentando desenvolver algo parecido com o que você quer e no fim quando fomos validar com os usuários, muitos deles reclamaram, o que me gerou estranheza à primeira vista mas depois que eu compreendi o contexto fez todo sentido.

Um usuário de leitor de tela normalmente aprende vários comandos (atalhos) do software pra facilitar a sua navegação nos sites. Se a gente for pensar de um modo geral, não há um padrão universal de acessibilidade na web (ou pelo menos não há um que TODOS os sites atuais utilizem) e isso gera um grande problema que as vezes nos passa despercebido: Cada site proporciona uma experiência diferente de navegação.

Diante disso, é mais fácil pra um usuário de leitor de tela aprender os atalhos do software do que ter que reaprender atalhos em cada site que ele visita. Compreende?

Por mais que a sua solução seja bacana, ela vai rodar apenas no teu site e, pensando em um usuário desse perfil, isso seria frustrante já que ele teria de aprender novos atalhos para navegar no teu site.

E esse é o momento em que a gente para e pensa: "Poxa vida... e agora?"
Acredito que o ideal seja manter sempre um bom diálogo próximo dos usuários, tentando compreender a fundo as suas necessidades.

De qualquer forma, te desejo sorte.
Que você consiga encontrar uma solução próxima da sua necessidade e principalmente das necessidades dos teus usuários.

PS: E se você desenvolver essa ferramenta, não deixe de compartilhar... Quem sabe um dia ela não vira um "padrão" na web.

**Comment by zemariobandeira** [comment]
@deppbrazil teve um experimento no google chrome que usava javascript para entregar áudio para os usuários, como um "menu de navegação por áudio". O site da Croacia Áudio chegou a implementar, mas já retiraram do ar, ao que me parece. https://experiments.withgoogle.com/chrome/croacia-audio

Bom, o facebook possui um recurso de acessibilidade interessante que talvez te interesse.
Basta abrir o seu facebook e digitar ALT + /  ou ALT + ; (dependendo da conf. do seu teclado) que irá abrir uma barra de acessibilidade no topo do site. Nela é possível navegar entre diferentes seções (via atributo ID) e escolher uma delas. Pra o seu caso, bastava adicionar um áudio explicativo para cada seção. Acredito que daria um bom efeito.

Qualquer coisa é só falar
Abraço

**Comment by deppbrazil** [comment]
@MarioDKR owww show mano.... valew vou dar um pesquisada mais afundo sim... 

### valew pelas dicas =)
